e2e/driver/ContainerInitializer.ts

```diff
@@ -0,0 +1,77 @@
+import { MultiUserLoginPage } from '../pageobjects/login/MultiUserLoginPage';
+import { TestConstants } from '../TestConstants';
+import { OcpLoginPage } from '../pageobjects/openshift/OcpLoginPage';
+import { OcpWebConsolePage } from '../pageobjects/openshift/OcpWebConsolePage';
+import { OcpLoginByTempAdmin } from '../pageobjects/login/OcpLoginByTempAdmin';
+import { ITestWorkspaceUtil } from '..';
+
+export function getContainer(): Container {
+    const e2eContainer = new Container();
+
+    e2eContainer.bind<IDriver>(TYPES.Driver).to(ChromeDriver).inSingletonScope();
+    e2eContainer.bind<ITestWorkspaceUtil>(TYPES.WorkspaceUtil).to(TestWorkspaceUtil).inSingletonScope();
+    e2eContainer.bind<IOcpLoginPage>(TYPES.OcpLogin).to(OcpLoginByTempAdmin).inSingletonScope();
+
+    if (TestConstants.TS_SELENIUM_MULTIUSER) {
+        e2eContainer.bind<ICheLoginPage>(TYPES.CheLogin).to(MultiUserLoginPage).inSingletonScope();
+    } else {
+        e2eContainer.bind<ICheLoginPage>(TYPES.CheLogin).to(SingleUserLoginPage).inSingletonScope();
+    }
+
+    e2eContainer.bind<DriverHelper>(CLASSES.DriverHelper).to(DriverHelper).inSingletonScope();
+    e2eContainer.bind<Dashboard>(CLASSES.Dashboard).to(Dashboard).inSingletonScope();
+    e2eContainer.bind<Workspaces>(CLASSES.Workspaces).to(Workspaces).inSingletonScope();
+    e2eContainer.bind<NewWorkspace>(CLASSES.NewWorkspace).to(NewWorkspace).inSingletonScope();
+    e2eContainer.bind<WorkspaceDetails>(CLASSES.WorkspaceDetails).to(WorkspaceDetails).inSingletonScope();
+    e2eContainer.bind<WorkspaceDetailsPlugins>(CLASSES.WorkspaceDetailsPlugins).to(WorkspaceDetailsPlugins).inSingletonScope();
+    e2eContainer.bind<Ide>(CLASSES.Ide).to(Ide).inSingletonScope();
+    e2eContainer.bind<ProjectTree>(CLASSES.ProjectTree).to(ProjectTree).inSingletonScope();
+    e2eContainer.bind<Editor>(CLASSES.Editor).to(Editor).inSingletonScope();
+    e2eContainer.bind<TopMenu>(CLASSES.TopMenu).to(TopMenu).inSingletonScope();
+    e2eContainer.bind<QuickOpenContainer>(CLASSES.QuickOpenContainer).to(QuickOpenContainer).inSingletonScope();
+    e2eContainer.bind<PreviewWidget>(CLASSES.PreviewWidget).to(PreviewWidget).inSingletonScope();
+    e2eContainer.bind<GitHubPlugin>(CLASSES.GitHubPlugin).to(GitHubPlugin).inSingletonScope();
+    e2eContainer.bind<RightToolbar>(CLASSES.RightToolbar).to(RightToolbar).inSingletonScope();
+    e2eContainer.bind<Terminal>(CLASSES.Terminal).to(Terminal).inSingletonScope();
+    e2eContainer.bind<DebugView>(CLASSES.DebugView).to(DebugView).inSingletonScope();
+    e2eContainer.bind<WarningDialog>(CLASSES.WarningDialog).to(WarningDialog).inSingletonScope();
+    e2eContainer.bind<ScreenCatcher>(CLASSES.ScreenCatcher).to(ScreenCatcher).inSingletonScope();
+    e2eContainer.bind<OcpLoginPage>(CLASSES.OcpLoginPage).to(OcpLoginPage).inSingletonScope();
+    e2eContainer.bind<OcpWebConsolePage>(CLASSES.OcpWebConsolePage).to(OcpWebConsolePage).inSingletonScope();
+    return e2eContainer;
```

Comment:
What a goal of these changes?

Reply:
Previously, all this stuff was in inversify.config.ts. Once we run our tests, this upstream inversify.config.ts was run, then our inversify.config.ts was run and we tried to rebind some of these dependencies. We exported upstream container, do the changes to the container and then use that container in our test. The problem is, that in that time, there were two containers - one upstream and one ours. Basically, the export we did was not exporting the container, but some copy of that probably. So when we were running our class and explicitly saying "use our container" it works (as here: https://github.com/redhat-developer/rh-che/blob/master/e2e-saas/tests/HappyPath.spec.ts#L24). But when some upstream class was executed, it injected upstream container and its setting was not affected by changes we did. So we need to have a mechanism to overwrite the upstream container - that means we need to overwrite the content of inversify.config.ts. 
Now, inversify.config.ts has a dynamic import of an ContainerInitializes.ts. It works that way -  the command ` let pathh = path.resolve('.'); ` gets the path of a root folder from where the tests are executed (for you that's che/e2e, for us it is rh-che/e2e-saas). It searches for file `driver/ContainerIntializer.ts` on that path and then injects the definition. You can see how our ContainerIntializer.ts looks like on our side here https://github.com/redhat-developer/rh-che/pull/1563/files#diff-b55da17e9ab25bab72c15a7cad5b9933R16. We are using the upstream definition and rebinding only those classes we need.
I hope it is clearer for you now. If you have any question, please feel free to ask, I know it is not a trivial logic.

Reply:
Yes, I understand now why you do that. But I am -1 about moving to the "driver" folder because I don't understand how the "inversify" config file related to driver. Also I am -1 about name "ContainerInitializer" because it may lead to a mess (for example when I see "Container" I think about docker). I guess you should left understandable name "inversify.config.ts" just reconfigure script on your side for searching by pattern "inversify.config.ts", and also left it in the root.

Reply:
I agree the naming should be better, I'm bad on that - so what would you suggest? And I agree with moving that file to root folder.
I don't understand the second half of your answer - what do you mean by `reconfigure script on your side for searching by pattern ` which script are you talking about?

Reply:
I mean this part:
```
It works that way - the command let pathh = path.resolve('.'); gets the path of a root folder from where the tests are executed (for you that's che/e2e, for us it is rh-che/e2e-saas). It searches for file driver/ContainerIntializer.ts on that path and then injects the definition.
```
and I was talking about changing "ContainerInitializer" in this logic to "inversify.config.ts"

Reply:
Sorry, I still can not get that. that path.resolve command is in your inversify.confit.ts file. https://github.com/eclipse/che/pull/14246/files#diff-123fa9ffe8b46ab5110d022438d584f4L42

Reply:
Nevermind, just keep the "inversify.conf.ts" name and keep placing it in the root directory

Reply:
There is still `inversify.config.ts` file which is placed in a root, as you can see here: https://github.com/eclipse/che/pull/14246/files#diff-123fa9ffe8b46ab5110d022438d584f4L1, but there is one more file - ContainerInitlizer.ts with extracted logic from inversify.config.ts. So maybe I could rename it to InversifyContianerInitializer.ts?

Reply:
Sorry but I don't understand why you can't reuse existed "inversify.config.ts"?

Reply:
We have discussed that on a call and decided to use the solution which is provided by this PR.